packages/flake-tracker/src/manager.js

```diff
@@ -11,34 +11,54 @@ class FlakeManager {
       )
     }
     this.github = github
+    this.setup = true
   }
 
   // Called at the beginning of each test suite
   static async build() {
-    db.init()
-    let github
-    if (shouldUseGitHub) {
-      github = await GitHub.build()
-      if (shouldSkipKnownFlakes) {
-        db.writeKnownFlakes(await github.fetchKnownFlakesToSkip())
+    try {
+      db.init()
+      let github
+      if (shouldUseGitHub) {
+        github = await GitHub.build()
+        if (shouldSkipKnownFlakes) {
+          db.writeKnownFlakes(
+            await github.fetchKnownFlakesToSkip().catch(() => {
+              console.log('failed to fetch list of known flakey tests')
+              return []
+            })
+          )
+        }
       }
+      return new FlakeManager(github)
+    } catch (error) {
+      this.setup = false
+      console.log('Flake tracker setup failed')
+      console.log(error)
```

Comment:
Maybe I'm missing something, but this build is assigned to a global var, and then something calls `finish()`
In this case, you are not returning anything, so probably that finish will fail.
If I'm not misunderstanding this, you could do 2 things. Re raise the error here. Or, send a new FlakeManager, that only has that flag set (you can add a default flag to the constructor to force skipping the github thing, and just set the flag as false.

Reply:
That's why we set this.setup = false  here so that finish() can return immediately when that happens

Reply:
Great catch!